packages/modal-ui/src/lib/components/DerivationPath.tsx

```diff
@@ -24,10 +24,27 @@ export const DerivationPath: React.FC<DerivationPathProps> = ({
   params,
   onError,
 }) => {
-  const [derivationPath, setDerivationPath] = useState(DEFAULT_DERIVATION_PATH);
+  const [derivationPaths, setDerivationPaths] = useState<
+    Array<{ path: string }>
+  >([{ path: DEFAULT_DERIVATION_PATH }]);
 
-  const handleDerivationPathChange = (e: ChangeEvent<HTMLInputElement>) => {
-    setDerivationPath(e.target.value);
+  const handleDerivationPathAdd = () => {
+    setDerivationPaths([...derivationPaths, { path: "" }]);
+  };
+
+  const handleDerivationPathRemove = (index: number) => {
+    const newPaths = [...derivationPaths];
+    newPaths.splice(index, 1);
+    setDerivationPaths(newPaths);
+  };
+
+  const handleDerivationPathChange = (
+    index: number,
+    e: ChangeEvent<HTMLInputElement>
+  ) => {
+    const newPaths = [...derivationPaths];
+    newPaths[index].path = e.target.value;
+    setDerivationPaths(newPaths);
```

Comment:
When deriving state from previous state like this, we should use the function parameter for setState

Reply:
Thanks for the review, I had some free time and adjusted the code.